README.md

```diff
@@ -22,49 +34,156 @@ You can do it manually following the instructions from [rust-lang.org](https://w
+
+To provide wallet functions and oracle information be sure to run lotus-cli at first. You can find how to do it in **Lotus-CLI establishing**.
+
+Once all the needed  modules will be started, you can continue with miner configuration.
 
-## Development
-### Build cpp-filecoin
+1. Create miner work directory (f.e. `$mkdir miner`)
+2. Download miner executable here : `https://link_to_download` and put it in work directory
+   3.Copy proof parameters json file with command: `cp /<path to cpp-filecoin>/core/proofs/parameters.json MINER_REPO/proof-params.json`
+4. Run yor miner with ` --repo REPO --miner-repo MINER_REPO --owner PUBLIC_ADRESS --miner-api API_PORT --sector-size SECTOR_SIZE`
 
-First build will likely take long time. However, you can cache binaries to [hunter-binary-cache](https://github.com/soramitsu/hunter-binary-cache) or even download binaries from the cache in case someone has already compiled project with the same compiler. To this end, you need to set up two environment variables:
+## Advanced Building
+This approach allows to build our project with your modifications,
+but be sure that your realisation follows consensus rules.
+
+1. clone [funon-filecoin project](https://github.com/filecoin-project/cpp-filecoin "cpp-filecoin project")
+
+2. go to `./cpp-filecoin` directory
+
+3. run `$git submodule --recursive --update`
+
+4. To finish your building you may also want to add your GitHub token, In such case you need to set up two environment variables:
 ```
 GITHUB_HUNTER_USERNAME=<github account name>
 GITHUB_HUNTER_TOKEN=<github token>
 ```
-To generate github token follow the [instructions](https://help.github.com/en/github/authenticating-to-github/creating-a-personal-access-token-for-the-command-line). Make sure `read:packages` and `write:packages` permissions are granted (step 7 in instructions).
+To generate GitHub token follow the [instructions](https://help.github.com/en/github/authenticating-to-github/creating-a-personal-access-token-for-the-command-line). Make sure `read:packages` and `write:packages` permissions are granted.
 
-This project is can be built with
+5. At this stage you will need a build directory, which can be created with `mkdir BUILD_REPO`
+   
+6. Build cmake with the following command: `cmake . -B BUILD_DIR`
 
-```
-mkdir build && cd build
-cmake -DCLANG_TIDY=ON ..
-make -j
-```
+7. build targets: node - for node executable, miner_main- for miner executable.
+   example: `cmake --build BUILD_DIR --target node miner_main`
 
-It is suggested to build project with clang-tidy checks, however if you wish to omit clang-tidy step, you can use `cmake ..` instead.
+8. First build will likely take long time. However, you can cache binaries to [hunter-binary-cache].
 
-Tests can be run with: 
-```
-cd build
-ctest
-```
+9. It is also suggested to build project with clang-tidy checks, however if you wish to omit clang-tidy step, you can use `cmake ..` instead.
```

Comment:
```suggestion
```
may add `cmake . -B BUILD_DIR -D CLANG_TIDY=ON` example to step 6, if needed (does it even build with that flag?)